drivers/ph_oem/ph_oem.c

```diff
@@ -0,0 +1,446 @@
+#define ADDR (dev->params.addr)
+#define EN_PIN (dev->params.enable_pin)
+
+static int ph_oem_init_test(const ph_oem_t *dev);
+
+int ph_oem_init(ph_oem_t *dev, const ph_oem_params_t *params)
+{
+    assert(dev && params);
+
+    dev->params = *params;
+
+    return ph_oem_init_test(dev);
+}
+
+static int ph_oem_init_test(const ph_oem_t *dev)
+{
+    uint8_t reg_data;
+
+    i2c_acquire(I2C);
+
+    /* Register read test */
+    if (i2c_read_regs(I2C, ADDR, PH_OEM_REG_DEVICE_TYPE,
+                      &reg_data, 1, 0x0) < 0) {
+        DEBUG("\n[ph_oem debug] init - error: unable to read reg %x\n",
+              PH_OEM_REG_DEVICE_TYPE);
+
+        i2c_release(I2C);
+        return PH_OEM_NODEV;
+    }
+
+    /* Test if the device ID of the attached pH OEM sensor equals the
+     * value of the PH_OEM_REG_DEVICE_TYPE register
+     * */
+    if (reg_data != PH_OEM_DEVICE_TYPE_ID) {
+        DEBUG("\n[ph_oem debug] init - error: the attached device is not a pH OEM "
+              "Sensor. Read Device Type ID is: %i\n", reg_data);
+        i2c_release(I2C);
+        return PH_OEM_NOT_PH;
+    }
+    i2c_release(I2C);
+
+    /* Initilize enable gpio pin if it is defined */
+    if (dev->params.enable_pin == GPIO_UNDEF) {
+        return PH_OEM_OK;
+    }
+    if (gpio_init(EN_PIN, GPIO_OUT) < 0) {
+        DEBUG("\n[ph_oem] init - initializing enable gpio pin failed\n");
+        return PH_OEM_GPIO_INIT_ERR;
+    }
+    gpio_set(EN_PIN);
+
+    return PH_OEM_OK;
+}
+
+static int ph_oem_unlock_address_reg(ph_oem_t *dev)
+{
+    uint8_t reg_value;
+
+    i2c_acquire(I2C);
+
+    i2c_write_reg(I2C, ADDR, PH_OEM_REG_UNLOCK, 0x55, 0x0);
+    i2c_write_reg(I2C, ADDR, PH_OEM_REG_UNLOCK, 0xAA, 0x0);
+    /* if successfully unlocked the register will equal 0x00 */
+    i2c_read_reg(I2C, ADDR, PH_OEM_REG_UNLOCK, &reg_value, 0x0);
+
+    if (reg_value != 0x00) {
+        DEBUG("\n[ph_oem debug] Failed at unlocking I2C address register. \n");
+        i2c_release(I2C);
+        return PH_OEM_WRITE_ERR;
+    }
+    i2c_release(I2C);
+
+    return PH_OEM_OK;
+}
+
+int ph_oem_set_i2c_address(ph_oem_t *dev, uint8_t addr)
+{
+    if (ph_oem_unlock_address_reg(dev) != PH_OEM_OK) {
+        return PH_OEM_WRITE_ERR;
+    }
+
+    i2c_acquire(I2C);
+
+    if (i2c_write_reg(I2C, ADDR, PH_OEM_REG_ADDRESS, addr, 0x0) < 0) {
+        DEBUG("\n[ph_oem debug] Setting I2C address to %x failed\n", addr);
+        i2c_release(I2C);
+        return PH_OEM_WRITE_ERR;
+    }
+
+    dev->params.addr = addr;
+    i2c_release(I2C);
+
+    return PH_OEM_OK;
+}
+
+int ph_oem_enable_interrupt(ph_oem_t *dev, ph_oem_interrupt_pin_cb_t cb,
+                            void *arg, ph_oem_irq_option_t option)
+{
+    if (dev->params.interrupt_pin == GPIO_UNDEF) {
+        return PH_OEM_INTERRUPT_GPIO_UNDEF;
+    }
+
+    int gpio_flank = 0;
+    int gpio_mode = 0;
+
+    dev->arg = arg;
+    dev->cb = cb;
+
+    /* gpio_mode might be different for your microcontroller. Just GPIO_IN
```

Comment:
If this is MCU dependent, maybe the gpio_mode should be passed as a parameter?

Reply:
Added a new parameter to this function to pass the gpio_mode